### ISSUE TITLE
Cannot do graph sync once Graph Importer Task fails

### ISSUE THREAD

**Issue by abs51295** [openshiftio/openshift.io#3136]
Steps performed:

1. Invoked `bayesianFlow` with `EPV` and  `force_graph_sync=True`. 

2. The flow had already been invoked before with `EPV` and it failed at `GraphImporterTask` because of a gremlin endpoint error (in my case, but it can be anything). So it had the entry in the `analyses` table.

3. Now, while the initial invocation failed as mentioned in step 2, I did it again as mentioned in step 1 to force sync the `EPV` in the graph. The flow as expected starts from `initAnalysisFlow` but since the entry is already in the `analyses` table [this](https://github.com/fabric8-analytics/fabric8-analytics-worker/blob/master/f8a_worker/workers/init_analysis_flow.py#L47) condition evaluates to true and hence it removes `EVN` from the arguments received.

4. Now after that, `bayesianAnalysisFlow` is not called since [this](https://github.com/fabric8-analytics/fabric8-analytics-worker/blob/master/f8a_worker/dispatcher/flows/bayesianFlow.yml#L22) condition evaluates to `false`. 

5. For [GraphImporterTask](https://github.com/fabric8-analytics/fabric8-analytics-worker/blob/master/f8a_worker/dispatcher/flows/bayesianFlow.yml#L29) both the [conditions](https://github.com/fabric8-analytics/fabric8-analytics-worker/blob/master/f8a_worker/dispatcher/flows/bayesianFlow.yml#L32) evaluate to true and hence it is invoked. But as expected it fails this [assert](https://github.com/fabric8-analytics/fabric8-analytics-worker/blob/master/f8a_worker/workers/graph_importer.py#L30) and hence it doesn't go further for ingestion.

So now we have no way to forcefully ingest in the graph 😢

**Comment by abs51295** [comment]
cc: @msrb to have a look and change the severity if needed.

**Comment by miteshvp** [comment]
@abs51295 - very detailed. Thanks. @msrb - we should not pop the EPV when `force_graph_sync` is true. WDYT?

**Comment by msrb** [comment]
@abs51295 

Short version:
Good catch Aagam :+1: let's see how we can unblock you here.

Long version:
To be honest, I think this whole `force_graph_sync` thing should be removed from flows completely. We should clearly draw the line between S3 ingestion and graph. Flows should only care about analyzing packages and storing results in S3. And that's it, nothing more. Graph subsystem should be notified when new data land in S3, but that's about it. From that point it should be solely responsibility of the data-importer to make sure that the data reaches graph.

TL;DR: data-importer needs to be way smarter than it is now. I don't see any reason why we should run flows just to (again) notify data-importer that some data still need to be synced to graph. data-importer knows what went wrong the first time when sync failed, so it should be its responsibility to handle it/report it/retry it.

Flow definition are already quite a mess in my opinion.
